### DISCUSSION TITLE
I tried to read logs from my application and push them into Apache Skywalking using opentelemetry collector

### DISCUSSION THREAD

**Discussion by PiyushSharma2812** [apache/skywalking#11629] [Q&A]
Configuration 
Apache Skywalking version v9.4
OpenTelemetry-collector version v0.89.0  
Operationg system Windows 11 Pro

**This is my Otelcol config file**

receivers: 
    otlp: 
        protocols: 
            grpc:

processors: 
    batch: 
exporters: 
     otlp: 
        endpoint: "localhost:11800" 
        tls: 
           insecure: true 
     logging: 
         loglevel: debug

service: 
    pipelines: 
       metrics: 
            receivers: [otlp] 
            processors: [batch] 
            exporters: [logging,otlp]

**This is my skywalking configuration file**

receiver-sharing-server: selector: ${SW_RECEIVER_SHARING_SERVER:default} default:

# For HTTP server
restHost: ${SW_RECEIVER_SHARING_REST_HOST:0.0.0.0}
restPort: ${SW_RECEIVER_SHARING_REST_PORT:0}
restContextPath: ${SW_RECEIVER_SHARING_REST_CONTEXT_PATH:/}
restMaxThreads: ${SW_RECEIVER_SHARING_REST_MAX_THREADS:200}
restIdleTimeOut: ${SW_RECEIVER_SHARING_REST_IDLE_TIMEOUT:30000}
restAcceptQueueSize: ${SW_RECEIVER_SHARING_REST_QUEUE_SIZE:0}
httpMaxRequestHeaderSize: ${SW_RECEIVER_SHARING_HTTP_MAX_REQUEST_HEADER_SIZE:8192}

# For gRPC server
gRPCHost: ${SW_RECEIVER_GRPC_HOST:0.0.0.0}
gRPCPort: ${SW_RECEIVER_GRPC_PORT:0}
maxConcurrentCallsPerConnection: ${SW_RECEIVER_GRPC_MAX_CONCURRENT_CALL:0}
maxMessageSize: ${SW_RECEIVER_GRPC_MAX_MESSAGE_SIZE:0}
gRPCThreadPoolQueueSize: ${SW_RECEIVER_GRPC_POOL_QUEUE_SIZE:0}
gRPCThreadPoolSize: ${SW_RECEIVER_GRPC_THREAD_POOL_SIZE:0}
gRPCSslEnabled: ${SW_RECEIVER_GRPC_SSL_ENABLED:false}
gRPCSslKeyPath: ${SW_RECEIVER_GRPC_SSL_KEY_PATH:""}
gRPCSslCertChainPath: ${SW_RECEIVER_GRPC_SSL_CERT_CHAIN_PATH:""}
gRPCSslTrustedCAsPath: ${SW_RECEIVER_GRPC_SSL_TRUSTED_CAS_PATH:""}
authentication: ${SW_AUTHENTICATION:""}
receiver-otel: selector: ${SW_OTEL_RECEIVER:default} default: enabledHandlers: ${SW_OTEL_RECEIVER_ENABLED_HANDLERS:"oc,otlp,otlp-logs"} enabledOtelRules: ${SW_OTEL_RECEIVER_ENABLED_OTEL_RULES:"apisix,k8s/,istio-controlplane,vm,mysql/,postgresql/,oap,aws-eks/,windows,aws-s3/,aws-dynamodb/"}

**While running the opentelemetry collector using otel configuration file below is the command that i am using**

C:\otelcol.exe --config C:\opentelemetry-collector\otelcol\otelcol_configuration.yaml

Getting this error 2023-12-04T20:44:40.603+0530 warn zapgrpc/zapgrpc.go:195 [core] [Channel #1 SubChannel #2] grpc: addrConn.createTransport failed to connect to {Addr: "localhost:11800", ServerName: "localhost:11800", }. Err: connection error: desc = "transport: Error while dialing: dial tcp [::1]:11800: connectex: No connection could be made because the target machine actively refused it." {"grpc_log": true}

**Comment by wu-sheng** [comment]
Do you have the latest release with this YAML?

```yaml
receiver-otel:
  selector: ${SW_OTEL_RECEIVER:default}
  default:
    enabledHandlers: ${SW_OTEL_RECEIVER_ENABLED_HANDLERS:"otlp-metrics,otlp-logs"}
```

otlp logging is a new thing added. Because that, you could only use skywalking logging exporter in the otlp collector.

**Reply by PiyushSharma2812** [reply]
I am using Apache Skywalking version v9.4
OpenTelemetry-collector version v0.89.0

getting logs in this format 
127.0.0.1 - - [30/Nov/2023:14:39:09 +0530] "GET / HTTP/1.1" 404 751
0:0:0:0:0:0:0:1 - - [30/Nov/2023:14:46:41 +0530] "GET / HTTP/1.1" 404 751
0:0:0:0:0:0:0:1 - - [30/Nov/2023:14:46:43 +0530] "GET / HTTP/1.1" 404 751
0:0:0:0:0:0:0:1 - - [30/Nov/2023:14:46:43 +0530] "GET / HTTP/1.1" 404 751
0:0:0:0:0:0:0:1 - - [30/Nov/2023:14:47:07 +0530] "GET / HTTP/1.1" 404 751

I guess we need to use service name also in this logs format but don't know how to add into it.

how we can ingest it to skywaking?

**Reply by wu-sheng** [reply]
First of all, I mean 404 represents no available service on that. OTLP native log format is supported since 9.5.0 and works like this, https://skywalking.apache.org/docs/main/v9.7.0/en/setup/backend/log-otlp/

9.4.0 only supports OTEL with skywalking-log exporter.

Both OTEL and native log exporter in OTEL collector are relying on gRPC channel, rather than HTTP 1.1 Please setup the IP/port correctly.

For any information you needed, learn LAL to set all more context, https://skywalking.apache.org/docs/main/v9.7.0/en/concepts-and-designs/lal/

**Reply by PiyushSharma2812** [reply]
**I am using v9.4 OTEL with skywalking as exporter but throws error then i changed the opentelemetry configuration to this** 
receivers:
  otlp:
    protocols:
      grpc: {}

processors:
  batch:
exporters:
  otlp:
    endpoint: "localhost:11800"
  logging:
    loglevel: debug

service:
  pipelines:
    metrics:          
      receivers: [otlp]
      processors: [batch]
      exporters: [logging,otlp]
      
**getting this error**
      2023-12-07T12:28:54.068+0530    warn    zapgrpc/zapgrpc.go:195  [core] [Channel #1 SubChannel #2] grpc: addrConn.createTransport failed to connect to {Addr: "localhost:11800", ServerName: "localhost:11800", }. Err: connection error: desc = "transport: authentication handshake failed: tls: first record does not look like a TLS handshake"        {"grpc_log": true}


if i add this 
tls:
      insecure: true
      
then getting this 
2023-12-07T15:21:18.156+0530    info    service@v0.88.0/telemetry.go:84 Setting up own telemetry...
2023-12-07T15:21:18.157+0530    info    service@v0.88.0/telemetry.go:201        Serving Prometheus metrics      {"address": ":8888", "level": "Basic"}
2023-12-07T15:21:18.157+0530    info    exporter@v0.88.0/exporter.go:275        Deprecated component. Will be removed in future releases.       {"kind": "exporter", "data_type": "metrics", "name": "logging"}
2023-12-07T15:21:18.157+0530    warn    common/factory.go:68    'loglevel' option is deprecated in favor of 'verbosity'. Set 'verbosity' to equivalent value to preserve behavior.  {"kind": "exporter", "data_type": "metrics", "name": "logging", "loglevel": "debug", "equivalent verbosity level": "Detailed"}
2023-12-07T15:21:18.158+0530    info    service@v0.88.0/service.go:143  Starting otelcol...     {"Version": "0.88.0", "NumCPU": 8}
2023-12-07T15:21:18.158+0530    info    extensions/extensions.go:33     Starting extensions...
2023-12-07T15:21:18.158+0530    warn    internal@v0.88.0/warning.go:40  Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks    {"kind": "receiver", "name": "otlp", "data_type": "metrics", "documentation": "https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks"}
2023-12-07T15:21:18.158+0530    info    otlpreceiver@v0.88.0/otlp.go:83 Starting GRPC server    {"kind": "receiver", "name": "otlp", "data_type": "metrics", "endpoint": "0.0.0.0:4317"}
2023-12-07T15:21:18.159+0530    info    service@v0.88.0/service.go:169  Everything is ready. Begin running and processing data.

but no logs is there

**Reply by wu-sheng** [reply]
The error with TLS is you didn't set up OAP receiver with TLS opened or properly, Then the second seems to be normal logs.

If you can see `No service name found in resource attributes, discarding the log` in the OAP, it mean you received the logs, but we require for `service.name` in the OTLP log attributes. The OTLP collection supports setting that for this pipeline.

**Comment by abhishekacharya828** [comment]
![Screenshot from 2024-04-12 17-00-24](https://github.com/apache/skywalking/assets/134144944/3d75782f-bda6-47de-b54d-711dc0450b95)
![Screenshot from 2024-04-12 17-00-24](https://github.com/apache/skywalking/assets/134144944/b4b87de3-b9e3-4b6a-a766-3045d3f313a2)



I want to sent logs filebeat to kafka and kafka to skywalking server , when i push log filebeat to kafaka topic it send logs to the kafka topic successfully but skywalking does not show it in it UI.  

please help me ?

**Comment by abhishekacharya828** [comment]
![Screenshot from 2024-04-12 18-54-28](https://github.com/apache/skywalking/assets/134144944/e2cbd497-c536-472f-9971-777dbce931a3)

This is my skywalk kafka-fetcher configuration with kafka topic . please verify it

**Reply by wu-sheng** [reply]
Kakfa fetcher is for native log format, not OTLP compatible yet. Please read docs first. 

https://skywalking.apache.org/docs/main/latest/en/setup/backend/log-otlp/

Only mention one receiver in the doc.
